Add release notes for 0.291 (#24445)
prestodb/presto: presto-docs/src/main/sphinx/release/release-0.291.rst

* Add support for ORC metadata cache invalidation based on file modification time. `#24346 <https://github.com/prestodb/presto/pull/24346>`_
* Add support for ``ALTER VIEW RENAME TO`` operation, including the necessary infrastructure for connector implementations. `#23749 <https://github.com/prestodb/presto/pull/23749>`_
* Add support for ``BigInt`` data type in the SQL Client on Presto UI on supported browsers. See `compatibility <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#browser_compatibility>`_ for the supported browsers. `#24336 <https://github.com/prestodb/presto/pull/24336>`_
* Add support for the histogram statistic type. `#22365 <https://github.com/prestodb/presto/pull/22365>`_
* Add support for time type partitioning in the ORC file format for Iceberg. `#24091 <https://github.com/prestodb/presto/pull/24091>`_
* Add testing for partitioning using time type in Iceberg. `#24091 <https://github.com/prestodb/presto/pull/24091>`_

> * Added catalog and schema level access checks in USE statement. `#23882 <https://github.com/prestodb/presto/pull/23882>`_

Reviewer: ```suggestion
* Add catalog and schema level access checks in USE statement. `#23882 <https://github.com/prestodb/presto/pull/23882>`_
```

* Enable ``scale-writers`` by default. `#24107 <https://github.com/prestodb/presto/pull/24107>`_

> * Improve HTTP responses from async page transport servlet, by sanitizing the URI before use. `#24004 <https://github.com/prestodb/presto/pull/24004>`_

Reviewer: Please move this entry to between lines 18 and 19. 

The entries in each section should be in the order listed in the [Order of changes](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#order-of-changes) in the Release Notes Guidelines: 

Fix
Improve
Add
Replace
Rename
Remove
Upgrade/Downgrade. 

If you encounter an entry that does not start with these keywords (such as line 40 "Enable scale-writers") put those at the end of that section. 

Please apply this ordering to each section. I might miss an entry that needs to be moved to its correct location in its section, and I ask your help to review the order of entries in each section to catch entries that I might overlook.

* Update usage of MD5 to SHA256. `#23903 <https://github.com/prestodb/presto/pull/23903>`_

> Security Changes

Reviewer: Similar to my other ask of ordering entries within each section, please reorder the sections of the release notes to follow the [Order of sections](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#order-of-sections) list in the Release Notes Guidelines. 

General
Prestissimo (Native Execution)
Security
JDBC
Web UI
Connectors - Order the Connectors sections alphabetically, which I see they already are. 
Verifier
Resource Groups
SPI
Documentation

______________________________________
* Improve partitioned remote exchanges for wide data sets (more than 500 columns) to use row wise encoding. `#23929 <https://github.com/prestodb/presto/pull/23929>`_
* Add native plan checker to the native sidecar plugin and native endpoint for Velox plan conversion. `#23596 <https://github.com/prestodb/presto/pull/23596>`_
* Add session properties ``native_spill_prefixsort_enabled``, ``native_prefixsort_normalized_key_max_bytes``, and ``native_prefixsort_min_rows``. `#24043 <https://github.com/prestodb/presto/pull/24043>`_
* Add support for automatic scaling of writer threads for partitioned tables. Can be enabled with the ``native_execution_scale_partitioned_writer_threads_enabled`` session property. Native execution only. `#24155 <https://github.com/prestodb/presto/pull/24155>`_
* Remove the ``experimental.table-writer-merge-operator-enabled`` configuration property and the ``table_writer_merge_operator_enabled`` session property. `#24145 <https://github.com/prestodb/presto/pull/24145>`_
* Remove deprecated  ``native_query_trace_task_reg_exp session`` property from Prestissimo. `#24270 <https://github.com/prestodb/presto/pull/24270>`_

> * Remove ``query-data-cache-enabled-default`` configuration property, which is no longer needed as per-split fine-grained cache control has been introduced. `#24372 <https://github.com/prestodb/presto/pull/24372>`_

Reviewer 2: As here the PRs appear for both adding and removing the ``query-data-cache-enabled-default`` configuration property (#24076 and #24372). Would it make sense to remove both entries?

Author: Good catch! Let me remove them, thanks. @denodo-research-labs

* Upgrade avro to version 1.11.4 in response to `CVE-2024-47561 <https://github.com/advisories/GHSA-r7pg-v2c8-mfg3>`_. `#23943 <https://github.com/prestodb/presto/pull/23943>`_
* Upgrade grpc dependencies to version 1.68.0 in response to `CVE-2022-25647 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-25647>`_. `#24051 <https://github.com/prestodb/presto/pull/24051>`_
* Upgrade gson from version 2.8.9 to v2.11.0. `#24247 <https://github.com/prestodb/presto/pull/24247>`_
* Upgrade gson to version 2.8.9 in response to `CVE-2022-25647 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-25647>`_. `#24051 <https://github.com/prestodb/presto/pull/24051>`_
* Upgrade jetty dependencies to version 9.4.56.v20240826 in response to `CVE-2024-8184 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-8184>`_. `#24184 <https://github.com/prestodb/presto/pull/24184>`_
* Upgrade netty dependencies to version 4.1.115.Final in response to `CVE-2024-47535 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-47535>`_. `#24137 <https://github.com/prestodb/presto/pull/24137>`_

> Bigquery Connector Changes

Reviewer: ```suggestion
BigQuery Connector Changes
```

______________________
* Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. `#23401 <https://github.com/prestodb/presto/pull/23401>`_

Iceberg Connector Changes
_________________________
* Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. `#23401 <https://github.com/prestodb/presto/pull/23401>`_
* Add configuration property ``iceberg.rest.nested.namespace.enabled`` to support nested namespaces in Iceberg's REST Catalog. Defaults to ``true``. `#24083 <https://github.com/prestodb/presto/pull/24083>`_
* Add support for ``ALTER VIEW RENAME TO``. `#23749 <https://github.com/prestodb/presto/pull/23749>`_
* Add support of ``view`` for Iceberg connector when configured with ``REST`` and ``NESSIE``. `#23793 <https://github.com/prestodb/presto/pull/23793>`_
* Add support of specifying table location on creation for Iceberg connector when configured with ``REST`` and ``NESSIE``. `#24218 <https://github.com/prestodb/presto/pull/24218>`_
* Remove in-memory hive metastore cache in Iceberg connector. `#24326 <https://github.com/prestodb/presto/pull/24326>`_

> * Optimize redundant getTable calls in Iceberg Connector. `#24376 <https://github.com/prestodb/presto/pull/24376>`_

Reviewer: ```suggestion
* Improve redundant getTable calls in Iceberg Connector. `#24376 <https://github.com/prestodb/presto/pull/24376>`_
```
Please move this revised entry to its correct location using the [Order of changes](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#order-of-changes) in the Release Notes Guidelines.

Memory Connector Changes
________________________
* Add support for ``ALTER VIEW RENAME TO``. `#23749 <https://github.com/prestodb/presto/pull/23749>`_

> Mongodb Connector Changes

Reviewer: ```suggestion
MongoDB Connector Changes
```

* Modify the signature of ``PlanCheckerProviderFactory.create`` to pass in a map of configuration properties and replace ``SimplePlanFragmentSerde`` with a ``PlanCheckerProviderContext``. `#23955 <https://github.com/prestodb/presto/pull/23955>`_

Prestissimo (native Execution) Changes
______________________________________
* Improve partitioned remote exchanges for wide data sets (more than 500 columns) to use row wise encoding. `#23929 <https://github.com/prestodb/presto/pull/23929>`_
* Add native plan checker to the native sidecar plugin and native endpoint for Velox plan conversion. `#23596 <https://github.com/prestodb/presto/pull/23596>`_
* Add session properties ``native_spill_prefixsort_enabled``, ``native_prefixsort_normalized_key_max_bytes``, and ``native_prefixsort_min_rows``. `#24043 <https://github.com/prestodb/presto/pull/24043>`_
* Remove the ``experimental.table-writer-merge-operator-enabled`` configuration property and the ``table_writer_merge_operator_enabled`` session property. `#24145 <https://github.com/prestodb/presto/pull/24145>`_

> * Deprecated  ``native_query_trace_task_reg_exp session`` property from Prestissimo. `#24270 <https://github.com/prestodb/presto/pull/24270>`_

Reviewer: ```suggestion
* Remove deprecated  ``native_query_trace_task_reg_exp session`` property from Prestissimo. `#24270 <https://github.com/prestodb/presto/pull/24270>`_
```

> * Support automatic scaling of writer threads for partitioned tables. Can be enabled with the ``native_execution_scale_partitioned_writer_threads_enabled`` session property. Native execution only. `#24155 <https://github.com/prestodb/presto/pull/24155>`_

Reviewer: ```suggestion
* Add support for automatic scaling of writer threads for partitioned tables. Can be enabled with the ``native_execution_scale_partitioned_writer_threads_enabled`` session property. Native execution only. `#24155 <https://github.com/prestodb/presto/pull/24155>`_
```
Please move this revised entry to its correct location using the [Order of changes](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#order-of-changes) in the Release Notes Guidelines.

* Add configuration property ``iceberg.rest.nested.namespace.enabled`` to support nested namespaces in Iceberg's REST Catalog. Defaults to ``true``. `#24083 <https://github.com/prestodb/presto/pull/24083>`_
* Add support for ``ALTER VIEW RENAME TO``. `#23749 <https://github.com/prestodb/presto/pull/23749>`_
* Add support of ``view`` for Iceberg connector when configured with ``REST`` and ``NESSIE``. `#23793 <https://github.com/prestodb/presto/pull/23793>`_
* Add support of specifying table location on creation for Iceberg connector when configured with ``REST`` and ``NESSIE``. `#24218 <https://github.com/prestodb/presto/pull/24218>`_
* Remove in-memory hive metastore cache in Iceberg connector. `#24326 <https://github.com/prestodb/presto/pull/24326>`_

Memory Connector Changes
________________________
* Add support for ``ALTER VIEW RENAME TO``. `#23749 <https://github.com/prestodb/presto/pull/23749>`_

MongoDB Connector Changes
_________________________
* Add steps to connect to MongoDB cluster with TLS CA File to :doc:`/connector/mongodb`. `#24352 <https://github.com/prestodb/presto/pull/24352>`_

> Clickhouse Connector Changes

Reviewer: Please move the heading, format line, and single entry for `Clickhouse Connector Changes` up (I think to line 94) so that this connector heading and content is in alphabetic order between `Cassandra Connector Changes` and `Delta Connector Changes`.

Reviewer 3: fixed

> ----------------------------

Reviewer: ```suggestion
____________________________
```
Please change this formatting line from hyphens (-) to underline (_) characters.

Reviewer 3: fixed

=============
Release 0.291
=============

**Highlights**
==============

**Details**
===========

General Changes
_______________

> * Fix behavior of :func:`width_bucket(x, bins) -> bigint` which previously treated all ``null`` elements in bins as ``0``. Now the function will throw an error if it finds a ``null`` or non-finite element in ``bins``..  :pr:`24103`. (`#24103 <https://github.com/prestodb/presto/pull/24103>`_)

Author: Created the issue https://github.com/prestodb/presto-release-tools/issues/41 in presto-release-tools to update the code generation

* Fix bug in ``strrpos`` function for multibyte characters. `#24226 <https://github.com/prestodb/presto/pull/24226>`_
* Improve Request Headers in the Authentication Filter Class. `#23380 <https://github.com/prestodb/presto/pull/23380>`_
* Improve coordinator task management performance. `#24369 <https://github.com/prestodb/presto/pull/24369>`_
* Improve efficiency of coordinator when running a large number of tasks. `#24288 <https://github.com/prestodb/presto/pull/24288>`_
* Improve scheduling for CTE materialization: Now, only the stages containing CTE table scans that reference CTE table write stages are blocked till the write is complete, instead of the entire query being blocked as was the case previously. This is controlled by the session property ``enhanced_cte_scheduling_enabled`` (on by default). `#24108 <https://github.com/prestodb/presto/pull/24108>`_

> * Add :doc:`/sql/alter-table` SET PROPERTIES statement `#21495 <https://github.com/prestodb/presto/pull/21495>`_

Reviewer: ```suggestion
* Add :doc:`/sql/alter-table` SET PROPERTIES statement. `#21495 <https://github.com/prestodb/presto/pull/21495>`_
```

* Add :func:`google_polyline_decode` function to convert Google polyline to Presto ST_Geometry types. `#23999 <https://github.com/prestodb/presto/pull/23999>`_
* Add :func:`google_polyline_encode` function to convert Presto ST_Geometry to Google polyline types. `#23999 <https://github.com/prestodb/presto/pull/23999>`_
* Add Delete TableWriter TableFinish node to SPI. `#24088 <https://github.com/prestodb/presto/pull/24088>`_
* Add SemiJoin Join TableWriter Delete TableFinish node to connector optimizer. `#24154 <https://github.com/prestodb/presto/pull/24154>`_
* Add ``ClientRequestFilter.java`` interface in Presto-spi. `#23380 <https://github.com/prestodb/presto/pull/23380>`_

> * Add ``query-data-cache-enabled-default`` configuration property to align C++ cache behavior with Java. Set it to ``true``(default) for current C++ behavior or to ``false`` to match Java's cache logic. `#24076 <https://github.com/prestodb/presto/pull/24076>`_

Reviewer: ```suggestion
* Add ``query-data-cache-enabled-default`` configuration property to align C++ cache behavior with Java. Set it to ``true`` (default) for current C++ behavior or to ``false`` to match Java's cache logic. `#24076 <https://github.com/prestodb/presto/pull/24076>`_
```

No space between the formatting tics for true and `(default)` results in this
<img width="774" alt="Screenshot 2025-01-30 at 10 10 48 AM" src="https://github.com/user-attachments/assets/c7d71bac-a989-4263-ae0b-97a41704298c" />

* Add a configuration property ``plan-checker.config-dir`` to set the configuration directory for PlanCheckerProvider configurations. `#23955 <https://github.com/prestodb/presto/pull/23955>`_

> * Add a session property `include_values_node_in_connector_optimizer` to enable connector optimizer optimize plans with values node  `#24227 <https://github.com/prestodb/presto/pull/24227>`_

Reviewer: ```suggestion
* Add a session property ``include_values_node_in_connector_optimizer`` to enable connector optimizer optimize plans with values node.  `#24227 <https://github.com/prestodb/presto/pull/24227>`_
```

* Add an optional input distribution constraint to DeleteNode. `#24104 <https://github.com/prestodb/presto/pull/24104>`_
* Add ``native_enforce_join_build_input_partition`` session property to not enforce input partition for join build. `#24163 <https://github.com/prestodb/presto/pull/24163>`_
* Add catalog and schema level access checks in :doc:`/sql/use` statement. `#24182 <https://github.com/prestodb/presto/pull/24182>`_
* Add delete node in subfield pruning optimizer. `#24206 <https://github.com/prestodb/presto/pull/24206>`_

> * Add single worker execution. To improve latency of tiny queries running on a large cluster, we introduce single worker execution mode: query will only use one node to execute and plan would be optimized accordingly. This feature can be turned on by config `single-node-execution-enabled` or session property `single_node_execution_enabled`. `#24172 <https://github.com/prestodb/presto/pull/24172>`_

Reviewer: ```suggestion
* Add single worker execution. To improve latency of tiny queries running on a large cluster, we introduce single worker execution mode: query will only use one node to execute and plan would be optimized accordingly. This feature can be turned on by the configuration property ``single-node-execution-enabled`` or the session property ``single_node_execution_enabled``. `#24172 <https://github.com/prestodb/presto/pull/24172>`_
```

* Add support for ``ALTER VIEW RENAME TO`` operation, including the necessary infrastructure for connector implementations. `#23749 <https://github.com/prestodb/presto/pull/23749>`_
* Add support for ``BigInt`` data type in the SQL Client on Presto UI on supported browsers. See `compatibility <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#browser_compatibility>`_ for the supported browsers. `#24336 <https://github.com/prestodb/presto/pull/24336>`_
* Add support for the histogram statistic type. `#22365 <https://github.com/prestodb/presto/pull/22365>`_
* Add support for time type partitioning in the ORC file format for Iceberg. `#24091 <https://github.com/prestodb/presto/pull/24091>`_
* Add testing for partitioning using time type in Iceberg. `#24091 <https://github.com/prestodb/presto/pull/24091>`_
* Added catalog and schema level access checks in USE statement. `#23882 <https://github.com/prestodb/presto/pull/23882>`_
* Remove ``query-data-cache-enabled-default`` configuration property, which is no longer needed as per-split fine-grained cache control has been introduced. `#24372 <https://github.com/prestodb/presto/pull/24372>`_

> * Remove the ``experimental.table-writer-merge-operator-enabled` config property and the ``table_writer_merge_operator_enabled`` session property. `#24145 <https://github.com/prestodb/presto/pull/24145>`_

Reviewer: ```suggestion
* Remove the ``experimental.table-writer-merge-operator-enabled`` configuration property and the ``table_writer_merge_operator_enabled`` session property. `#24145 <https://github.com/prestodb/presto/pull/24145>`_
```

> * Deprecated  native_query_trace_task_reg_exp session property from Prestissimo. `#24270 <https://github.com/prestodb/presto/pull/24270>`_

Reviewer: ```suggestion
* Deprecated  ``native_query_trace_task_reg_exp session`` property from Prestissimo. `#24270 <https://github.com/prestodb/presto/pull/24270>`_
```

> * Deprecated  native_query_trace_task_reg_exp session property from Prestissimo. `#24270 <https://github.com/prestodb/presto/pull/24270>`_

Reviewer: Please move this entry from _General Changes_ to _Prestissimo (native Execution) Changes_.

> * Remove ``query-data-cache-enabled-default`` configuration property, which is no longer needed as per-split fine-grained cache control has been introduced. :pr:`24372 `. (`#24372 <https://github.com/prestodb/presto/pull/24372>`_)

Reviewer: ```suggestion
* Remove ``query-data-cache-enabled-default`` configuration property, which is no longer needed as per-split fine-grained cache control has been introduced. `#24372 <https://github.com/prestodb/presto/pull/24372>`_
```

* Enable ``scale-writers`` by default. `#24107 <https://github.com/prestodb/presto/pull/24107>`_

> * Support automatic scaling of writer threads for partitioned tables. Can be enabled with the `native_execution_scale_partitioned_writer_threads_enabled` session property. Native execution only. `#24155 <https://github.com/prestodb/presto/pull/24155>`_

Reviewer: ```suggestion
* Support automatic scaling of writer threads for partitioned tables. Can be enabled with the ``native_execution_scale_partitioned_writer_threads_enabled`` session property. Native execution only. `#24155 <https://github.com/prestodb/presto/pull/24155>`_
```

> * Support automatic scaling of writer threads for partitioned tables. Can be enabled with the `native_execution_scale_partitioned_writer_threads_enabled` session property. Native execution only. `#24155 <https://github.com/prestodb/presto/pull/24155>`_

Reviewer: Please move this entry from _General Changes_ to _Prestissimo (native Execution) Changes_.

* Improve HTTP responses from async page transport servlet, by sanitizing the URI before use. `#24004 <https://github.com/prestodb/presto/pull/24004>`_
* Update usage of MD5 to SHA256. `#23903 <https://github.com/prestodb/presto/pull/23903>`_

> * Upgrade avro to version 1.11.4 in response to `CVE-2024-47561 <https://github.com/advisories/GHSA-r7pg-v2c8-mfg3>`_. `#23943 <https://github.com/prestodb/presto/pull/23943>`_

Reviewer: Please move the entries in lines 47-52 - the ones that say "in response to CVE" - from _General Changes_ to _Security Changes_.

> * Update usage of MD5 to SHA256 :pr:`23903`. (`#23903 <https://github.com/prestodb/presto/pull/23903>`_)

Reviewer: ```suggestion
* Update usage of MD5 to SHA256. `#23903 <https://github.com/prestodb/presto/pull/23903>`_
```

> * Upgrade avro to version 1.11.4 in response to `CVE-2024-47561 <https://github.com/advisories/GHSA-r7pg-v2c8-mfg3>`_. :pr:`23868`. (`#23943 <https://github.com/prestodb/presto/pull/23943>`_)

Reviewer: ```suggestion
* Upgrade avro to version 1.11.4 in response to `CVE-2024-47561 <https://github.com/advisories/GHSA-r7pg-v2c8-mfg3>`_. `#23943 <https://github.com/prestodb/presto/pull/23943>`_
```

> * Upgrade grpc dependencies to version 1.68.0 in response to `CVE-2022-25647 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-25647>` :pr:`24051`. (`#24051 <https://github.com/prestodb/presto/pull/24051>`_)

Reviewer: ```suggestion
* Upgrade grpc dependencies to version 1.68.0 in response to `CVE-2022-25647 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-25647>`_. `#24051 <https://github.com/prestodb/presto/pull/24051>`_
```

> * Upgrade gson from version 2.8.9 to v2.11.0 pr: `24247`. (`#24247 <https://github.com/prestodb/presto/pull/24247>`_)

Reviewer: ```suggestion
* Upgrade gson from version 2.8.9 to v2.11.0. `#24247 <https://github.com/prestodb/presto/pull/24247>`_
```

> * Upgrade gson to version 2.8.9 in response to `CVE-2022-25647 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-25647>` :pr:`24051`. (`#24051 <https://github.com/prestodb/presto/pull/24051>`_)

Reviewer: ```suggestion
* Upgrade gson to version 2.8.9 in response to `CVE-2022-25647 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-25647>`_. `#24051 <https://github.com/prestodb/presto/pull/24051>`_
```

> * Upgrade jetty dependencies to version 9.4.56.v20240826 in response to `CVE-2024-8184 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-8184>`_. :pr:`24184`. (`#24184 <https://github.com/prestodb/presto/pull/24184>`_)

Reviewer: ```suggestion
* Upgrade jetty dependencies to version 9.4.56.v20240826 in response to `CVE-2024-8184 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-8184>`_. `#24184 <https://github.com/prestodb/presto/pull/24184>`_
```

> * Upgrade netty dependencies to version 4.1.115.Final in response to `CVE-2024-47535 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-47535>`_. :pr:`24137`. (`#24137 <https://github.com/prestodb/presto/pull/24137>`_)

Reviewer: ```suggestion
* Upgrade netty dependencies to version 4.1.115.Final in response to `CVE-2024-47535 <https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-47535>`_. `#24137 <https://github.com/prestodb/presto/pull/24137>`_
```

Security Changes
________________

> * Fix `CVE-2015-3192 <https://www.mend.io/vulnerability-database/CVE-2015-3192>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2015-3192 <https://www.mend.io/vulnerability-database/CVE-2015-3192>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2015-5211 <https://www.mend.io/vulnerability-database/CVE-2015-5211>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2015-5211 <https://www.mend.io/vulnerability-database/CVE-2015-5211>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2016-1000027 <https://www.mend.io/vulnerability-database/CVE-2016-1000027>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2016-1000027 <https://www.mend.io/vulnerability-database/CVE-2016-1000027>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2018-1199 <https://www.mend.io/vulnerability-database/CVE-2018-1199>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2018-1199 <https://www.mend.io/vulnerability-database/CVE-2018-1199>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2018-1272 <https://www.mend.io/vulnerability-database/CVE-2018-1272>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2018-1272 <https://www.mend.io/vulnerability-database/CVE-2018-1272>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2020-5421 <https://www.mend.io/vulnerability-database/CVE-2020-5421>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2020-5421 <https://www.mend.io/vulnerability-database/CVE-2020-5421>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2021-0170 <https://www.mend.io/vulnerability-database/CVE-2021-0170>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2021-0170 <https://www.mend.io/vulnerability-database/CVE-2021-0170>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2021-22060 <https://www.mend.io/vulnerability-database/CVE-2021-22060>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2021-22060 <https://www.mend.io/vulnerability-database/CVE-2021-22060>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2021-22096 <https://www.mend.io/vulnerability-database/CVE-2021-22096>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2021-22096 <https://www.mend.io/vulnerability-database/CVE-2021-22096>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2021-22096 <https://www.mend.io/vulnerability-database/CVE-2021-22096>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2021-22096 <https://www.mend.io/vulnerability-database/CVE-2021-22096>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

* Fix `CVE-2022-22965 <https://www.mend.io/vulnerability-database/CVE-2022-22965>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
* Fix `CVE-2022-22970 <https://www.mend.io/vulnerability-database/CVE-2022-22970>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
* Fix `CVE-2022-22970 <https://www.mend.io/vulnerability-database/CVE-2022-22970>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
* Fix `CVE-2022-27772 <https://www.mend.io/vulnerability-database/CVE-2022-27772>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_

> * Fix `CVE-2023-20883 <https://www.mend.io/vulnerability-database/CVE-2023-20883>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2023-20883 <https://www.mend.io/vulnerability-database/CVE-2023-20883>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```
Missed this one yesterday in my first pass to clean up the redundant manual PR links.

> * Fix `CVE-2022-22965 <https://www.mend.io/vulnerability-database/CVE-2022-22965>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2022-22965 <https://www.mend.io/vulnerability-database/CVE-2022-22965>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2022-22970 <https://www.mend.io/vulnerability-database/CVE-2022-22970>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2022-22970 <https://www.mend.io/vulnerability-database/CVE-2022-22970>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2022-27772 <https://www.mend.io/vulnerability-database/CVE-2022-27772>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2022-27772 <https://www.mend.io/vulnerability-database/CVE-2022-27772>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2024-22243 <https://www.mend.io/vulnerability-database/CVE-2024-22243>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2024-22243 <https://www.mend.io/vulnerability-database/CVE-2024-22243>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2015-3192 <https://www.mend.io/vulnerability-database/CVE-2015-3192>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2015-5211 <https://www.mend.io/vulnerability-database/CVE-2015-5211>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2016-1000027 <https://www.mend.io/vulnerability-database/CVE-2016-1000027>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2018-1199 <https://www.mend.io/vulnerability-database/CVE-2018-1199>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2018-1272 <https://www.mend.io/vulnerability-database/CVE-2018-1272>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2020-5421 <https://www.mend.io/vulnerability-database/CVE-2020-5421>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2021-0170 <https://www.mend.io/vulnerability-database/CVE-2021-0170>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2021-22060 <https://www.mend.io/vulnerability-database/CVE-2021-22060>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2021-22096 <https://www.mend.io/vulnerability-database/CVE-2021-22096>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2021-22096 <https://www.mend.io/vulnerability-database/CVE-2021-22096>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2022-22965 <https://www.mend.io/vulnerability-database/CVE-2022-22965>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2022-22970 <https://www.mend.io/vulnerability-database/CVE-2022-22970>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2022-22970 <https://www.mend.io/vulnerability-database/CVE-2022-22970>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2022-27772 <https://www.mend.io/vulnerability-database/CVE-2022-27772>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2023-20883 <https://www.mend.io/vulnerability-database/CVE-2023-20883>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2024-22243 <https://www.mend.io/vulnerability-database/CVE-2024-22243>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2024-22259 <https://www.mend.io/vulnerability-database/CVE-2024-22259>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2024-22262 <https://www.mend.io/vulnerability-database/CVE-2024-22262>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2024-38809 <https://www.mend.io/vulnerability-database/CVE-2024-38809>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2024-6763 <https://www.mend.io/vulnerability-database/CVE-2024-6763>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2024-6763 <https://www.mend.io/vulnerability-database/CVE-2024-6763>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
> * Fix `CVE-2024-8184 <https://www.mend.io/vulnerability-database/CVE-2024-8184>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_

Reviewer 4: ```suggestion
```

Let's delete these, because it's a part of benchto which is not customer-facing.

> * Fix `CVE-2024-22259 <https://www.mend.io/vulnerability-database/CVE-2024-22259>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2024-22259 <https://www.mend.io/vulnerability-database/CVE-2024-22259>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2024-22262 <https://www.mend.io/vulnerability-database/CVE-2024-22262>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2024-22262 <https://www.mend.io/vulnerability-database/CVE-2024-22262>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2024-38809 <https://www.mend.io/vulnerability-database/CVE-2024-38809>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2024-38809 <https://www.mend.io/vulnerability-database/CVE-2024-38809>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

* Fix security vulnerability in presto-pinot-toolkit and presto-product-tests in response to `CVE-2020-0287 <https://nvd.nist.gov/vuln/detail/CVE-2020-0287>`_. `#24249 <https://github.com/prestodb/presto/pull/24249>`_

> * Fix security vulnerability in swagger-ui jar in response to 'CVE-2018-25031 <https://nvd.nist.gov/vuln/detail/CVE-2018-25031>' .  `#24153 <https://github.com/prestodb/presto/pull/24153>`_

Reviewer: ```suggestion
* Fix security vulnerability in swagger-ui jar in response to `CVE-2018-25031 <https://nvd.nist.gov/vuln/detail/CVE-2018-25031>`_.  `#24153 <https://github.com/prestodb/presto/pull/24153>`_
```
Corrected formatting of CVE link from ' to `.

> * Fix `CVE-2024-6763 <https://www.mend.io/vulnerability-database/CVE-2024-6763>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2024-6763 <https://www.mend.io/vulnerability-database/CVE-2024-6763>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix security vulnerability in swagger-ui jar in response to 'CVE-2018-25031 <https://nvd.nist.gov/vuln/detail/CVE-2018-25031>' . `#24199 <https://github.com/prestodb/presto/pull/24199>`_

Reviewer: ```suggestion
* Fix security vulnerability in swagger-ui jar in response to `CVE-2018-25031 <https://nvd.nist.gov/vuln/detail/CVE-2018-25031>`_. `#24199 <https://github.com/prestodb/presto/pull/24199>`_
```

> * Fix `CVE-2024-6763 <https://www.mend.io/vulnerability-database/CVE-2024-6763>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2024-6763 <https://www.mend.io/vulnerability-database/CVE-2024-6763>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix `CVE-2024-8184 <https://www.mend.io/vulnerability-database/CVE-2024-8184>`_. :pr:`24112`. (`#24112 <https://github.com/prestodb/presto/pull/24112>`_)

Reviewer: ```suggestion
* Fix `CVE-2024-8184 <https://www.mend.io/vulnerability-database/CVE-2024-8184>`_. `#24112 <https://github.com/prestodb/presto/pull/24112>`_
```

> * Fix security vulnerability in presto-pinot-toolkit and presto-product-tests in response to `CVE-2020-0287 <https://nvd.nist.gov/vuln/detail/CVE-2020-0287>`_. :pr:`24249`. (`#24249 <https://github.com/prestodb/presto/pull/24249>`_)

Reviewer: ```suggestion
* Fix security vulnerability in presto-pinot-toolkit and presto-product-tests in response to `CVE-2020-0287 <https://nvd.nist.gov/vuln/detail/CVE-2020-0287>`_. `#24249 <https://github.com/prestodb/presto/pull/24249>`_
```

> * Fix security vulnerability in swagger-ui jar in response to 'CVE-2018-25031 <https://nvd.nist.gov/vuln/detail/CVE-2018-25031>' .  :pr:`24153`. (`#24153 <https://github.com/prestodb/presto/pull/24153>`_)

Reviewer: ```suggestion
* Fix security vulnerability in swagger-ui jar in response to 'CVE-2018-25031 <https://nvd.nist.gov/vuln/detail/CVE-2018-25031>' .  `#24153 <https://github.com/prestodb/presto/pull/24153>`_
```

> * Fix security vulnerability in swagger-ui jar in response to 'CVE-2018-25031 <https://nvd.nist.gov/vuln/detail/CVE-2018-25031>' .  :pr:`24199`. (`#24199 <https://github.com/prestodb/presto/pull/24199>`_)

Reviewer: ```suggestion
* Fix security vulnerability in swagger-ui jar in response to 'CVE-2018-25031 <https://nvd.nist.gov/vuln/detail/CVE-2018-25031>' . `#24199 <https://github.com/prestodb/presto/pull/24199>`_
```

* Improve pbkdf2 hashing using SHA-256 cipher in response to `CWE-759 <https://cwe.mitre.org/data/definitions/759.htm>`_. `#24132 <https://github.com/prestodb/presto/pull/24132>`_
* Add re2j regex matching in QueryStateInfoResource to protect from ReDoS attacks in response to `CVE-2024-45296 <https://www.cve.org/CVERecord?id=CVE-2024-45296>`_. `#24048 <https://github.com/prestodb/presto/pull/24048>`_
* Add security-related headers to the static resources serving from the Presto UI, including: ``Content-Security-Policy``, ``X-Content-Type-Options``. See reference docs `Content-Security-Policy <https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP>`_ and `X-Content-Type-Options <https://learn.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/compatibility/gg622941(v=vs.85)>`_. `#24272 <https://github.com/prestodb/presto/pull/24272>`_

> * Add support for pluggable Custom Presto Authenticators `#24111 <https://github.com/prestodb/presto/pull/24111>`_

Reviewer: ```suggestion
* Add support for pluggable Custom Presto Authenticators. `#24111 <https://github.com/prestodb/presto/pull/24111>`_
```

> * Improve pbkdf2 hashing using SHA-256 cipher in response to `CWE-759 <https://cwe.mitre.org/data/definitions/759.htm>`_. :pr:`24132`. (`#24132 <https://github.com/prestodb/presto/pull/24132>`_)

Reviewer: ```suggestion
* Improve pbkdf2 hashing using SHA-256 cipher in response to `CWE-759 <https://cwe.mitre.org/data/definitions/759.htm>`_. `#24132 <https://github.com/prestodb/presto/pull/24132>`_
```

> * Add re2j regex matching in QueryStateInfoResource to protect from ReDoS attacks in response to `CVE-2024-45296 <https://www.cve.org/CVERecord?id=CVE-2024-45296>`_. :pr:`24048`. (`#24048 <https://github.com/prestodb/presto/pull/24048>`_)

Reviewer: ```suggestion
* Add re2j regex matching in QueryStateInfoResource to protect from ReDoS attacks in response to `CVE-2024-45296 <https://www.cve.org/CVERecord?id=CVE-2024-45296>`_. `#24048 <https://github.com/prestodb/presto/pull/24048>`_
```

* Replace `alluxio-shaded-client` with `alluxio-core` libraries in response to `CVE-2023-44981 <https://github.com/advisories/GHSA-7286-pgfv-vxvh>`_. `#24231 <https://github.com/prestodb/presto/pull/24231>`_

Bigquery Connector Changes
__________________________
* Fix for BigQuery ``SELECT``. `#23957 <https://github.com/prestodb/presto/pull/23957>`_

Cassandra Connector Changes
___________________________

> * Improve cryptographic protocol in response to `java:S4423 <https://sonarqube.ow2.org/coding_rules?open=java%3AS4423&rule_key=java%3AS4423>`_. `#24436 <https://github.com/prestodb/presto/pull/24436>`_

Reviewer: ```suggestion
* Improve cryptographic protocol in response to `Weak SSL/TLS protocols should not be used <https://sonarqube.ow2.org/coding_rules?open=java%3AS4423&rule_key=java%3AS4423>`_. `#24436 <https://github.com/prestodb/presto/pull/24436>`_
```

Suggest changing the link text from `java:S4423` to `Weak SSL/TLS protocols should not be used`, as it's good practice for a link to display in the link what the user will see when they click on the link. Also, the title of the destination page (Weak SSL/TLS protocols should not be used) is less cryptic than (java:S4423).

> * Add security-related headers to the static resources serving from the Presto UI, including: ``Content-Security-Policy``, ``X-Content-Type-Options``. See reference docs `Content-Security-Policy <https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP>`_ and `X-Content-Type-Options <https://learn.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/compatibility/gg622941(v=vs.85)>`_. :pr:`24272`. (`#24272 <https://github.com/prestodb/presto/pull/24272>`_)

Reviewer: ```suggestion
* Add security-related headers to the static resources serving from the Presto UI, including: ``Content-Security-Policy``, ``X-Content-Type-Options``. See reference docs `Content-Security-Policy <https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP>`_ and `X-Content-Type-Options <https://learn.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/compatibility/gg622941(v=vs.85)>`_. `#24272 <https://github.com/prestodb/presto/pull/24272>`_
```

> * Add support for pluggable Custom Presto Authenticators :pr:`#24111`. (`#24111 <https://github.com/prestodb/presto/pull/24111>`_)

Reviewer: ```suggestion
* Add support for pluggable Custom Presto Authenticators `#24111 <https://github.com/prestodb/presto/pull/24111>`_
```

> * Replace `alluxio-shaded-client` with `alluxio-core` libraries libraries in response to `CVE-2023-44981 <https://github.com/advisories/GHSA-7286-pgfv-vxvh>`_. :pr:`24231`. (`#24231 <https://github.com/prestodb/presto/pull/24231>`_)

Reviewer: ```suggestion
* Replace `alluxio-shaded-client` with `alluxio-core` libraries libraries in response to `CVE-2023-44981 <https://github.com/advisories/GHSA-7286-pgfv-vxvh>`_. `#24231 <https://github.com/prestodb/presto/pull/24231>`_
```

> * Fix for BigQuery ``SELECT``. :pr:`23957`. (`#23957 <https://github.com/prestodb/presto/pull/23957>`_)

Reviewer: ```suggestion
* Fix for BigQuery ``SELECT``. `#23957 <https://github.com/prestodb/presto/pull/23957>`_
```

> * Improve cryptographic protocol in response to `java:S4423 <https://sonarqube.ow2.org/coding_rules?open=java%3AS4423&rule_key=java%3AS4423>`_. :pr:`24436`. (`#24436 <https://github.com/prestodb/presto/pull/24436>`_)

Reviewer: ```suggestion
* Improve cryptographic protocol in response to `java:S4423 <https://sonarqube.ow2.org/coding_rules?open=java%3AS4423&rule_key=java%3AS4423>`_. `#24436 <https://github.com/prestodb/presto/pull/24436>`_
```

Delta Connector Changes
_______________________

> * Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. :pr:`23401`. (`#23401 <https://github.com/prestodb/presto/pull/23401>`_)

Reviewer: ```suggestion
* Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. `#23401 <https://github.com/prestodb/presto/pull/23401>`_
```

Hive Connector Changes
______________________

> * Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. :pr:`23401`. (`#23401 <https://github.com/prestodb/presto/pull/23401>`_)

Reviewer: ```suggestion
* Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. `#23401 <https://github.com/prestodb/presto/pull/23401>`_
```

Hudi Connector Changes
______________________

> * Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. :pr:`23401`. (`#23401 <https://github.com/prestodb/presto/pull/23401>`_)

Reviewer: ```suggestion
* Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. `#23401 <https://github.com/prestodb/presto/pull/23401>`_
```

Iceberg Connector Changes
_________________________

> * Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. :pr:`23401`. (`#23401 <https://github.com/prestodb/presto/pull/23401>`_)

Reviewer: ```suggestion
* Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. `#23401 <https://github.com/prestodb/presto/pull/23401>`_
```

> * Add configuration property ``iceberg.rest.nested.namespace.enabled`` to support nested namespaces in Iceberg's REST Catalog. Defaults to ``true``. :pr:`24083`. (`#24083 <https://github.com/prestodb/presto/pull/24083>`_)

Reviewer: ```suggestion
* Add configuration property ``iceberg.rest.nested.namespace.enabled`` to support nested namespaces in Iceberg's REST Catalog. Defaults to ``true``. `#24083 <https://github.com/prestodb/presto/pull/24083>`_
```

> * Add support for ``ALTER VIEW RENAME TO``. :pr:`23749`. (`#23749 <https://github.com/prestodb/presto/pull/23749>`_)

Reviewer: ```suggestion
* Add support for ``ALTER VIEW RENAME TO``. `#23749 <https://github.com/prestodb/presto/pull/23749>`_
```

> * Add support of ``view`` for Iceberg connector when configured with ``REST`` and ``NESSIE``. :pr:`23793`. (`#23793 <https://github.com/prestodb/presto/pull/23793>`_)

Reviewer: ```suggestion
* Add support of ``view`` for Iceberg connector when configured with ``REST`` and ``NESSIE``. `#23793 <https://github.com/prestodb/presto/pull/23793>`_
```

> * Add support of specifying table location on creation for Iceberg connector when configured with ``REST`` and ``NESSIE``. :pr:`24218`. (`#24218 <https://github.com/prestodb/presto/pull/24218>`_)

Reviewer: ```suggestion
* Add support of specifying table location on creation for Iceberg connector when configured with ``REST`` and ``NESSIE``. `#24218 <https://github.com/prestodb/presto/pull/24218>`_
```

> * Remove in-memory hive metastore cache in Iceberg connector. :pr:`24326`. (`#24326 <https://github.com/prestodb/presto/pull/24326>`_)

Reviewer: ```suggestion
* Remove in-memory hive metastore cache in Iceberg connector. `#24326 <https://github.com/prestodb/presto/pull/24326>`_
```

> * Optimize redundant getTable calls in Iceberg Connector :pr:`24376 `. (`#24376 <https://github.com/prestodb/presto/pull/24376>`_)

Reviewer: ```suggestion
* Optimize redundant getTable calls in Iceberg Connector. `#24376 <https://github.com/prestodb/presto/pull/24376>`_
```

Memory Connector Changes
________________________

> * Add support for ``ALTER VIEW RENAME TO``. :pr:`23749`. (`#23749 <https://github.com/prestodb/presto/pull/23749>`_)

Reviewer: ```suggestion
* Add support for ``ALTER VIEW RENAME TO``. `#23749 <https://github.com/prestodb/presto/pull/23749>`_
```

Mongodb Connector Changes
_________________________

> * Add steps to connect to MongoDB cluster with TLS CA File to :doc:`/connector/mongodb`. :pr:`24352`. (`#24352 <https://github.com/prestodb/presto/pull/24352>`_)

Reviewer: ```suggestion
* Add steps to connect to MongoDB cluster with TLS CA File to :doc:`/connector/mongodb`. `#24352 <https://github.com/prestodb/presto/pull/24352>`_
```

SPI Changes
___________

> * Improve ExpressionOptimizer#optimize method with a variable resolver to return ``RowExpression``. :pr:`24287`. (`#24287 <https://github.com/prestodb/presto/pull/24287>`_)

Reviewer: ```suggestion
* Improve ExpressionOptimizer#optimize method with a variable resolver to return ``RowExpression``. `#24287 <https://github.com/prestodb/presto/pull/24287>`_
```

> * Add WindowNode, JoinNode, SemiJoinNode, MergeJoinNode, and SpatialJoinNode to the SPI. :pr:`23976`. (`#23976 <https://github.com/prestodb/presto/pull/23976>`_)

Reviewer: ```suggestion
* Add WindowNode, JoinNode, SemiJoinNode, MergeJoinNode, and SpatialJoinNode to the SPI. `#23976 <https://github.com/prestodb/presto/pull/23976>`_
```

* Add a partition by attribute to specify the scope of sort node. `#24095 <https://github.com/prestodb/presto/pull/24095>`_
* Remove ConnectorJoinNode from the SPI. JoinNode can now be used instead. `#23976 <https://github.com/prestodb/presto/pull/23976>`_
* Remove experimental getPreferredShuffleLayout methods from the connector metadata in favor of existing `getNewTableLayout`, `getInsertLayout` methods `#24106 <https://github.com/prestodb/presto/pull/24106>`_
* Modify the signature of ``PlanCheckerProviderFactory.create`` to pass in a map of configuration properties and replace ``SimplePlanFragmentSerde`` with a ``PlanCheckerProviderContext``. `#23955 <https://github.com/prestodb/presto/pull/23955>`_

Prestissimo (native Execution) Changes
______________________________________
* Improve partitioned remote exchanges for wide data sets (more than 500 columns) to use row wise encoding. `#23929 <https://github.com/prestodb/presto/pull/23929>`_
* Add native plan checker to the native sidecar plugin and native endpoint for Velox plan conversion. `#23596 <https://github.com/prestodb/presto/pull/23596>`_

> * Add session properties ``native_spill_prefixsort_enabled``, ``native_prefixsort_normalized_key_max_bytes ``, and ``native_prefixsort_min_rows ``. `#24043 <https://github.com/prestodb/presto/pull/24043>`_

Reviewer: ```suggestion
* Add session properties ``native_spill_prefixsort_enabled``, ``native_prefixsort_normalized_key_max_bytes``, and ``native_prefixsort_min_rows``. `#24043 <https://github.com/prestodb/presto/pull/24043>`_
```

The gap after the second and third property names and the `` breaks their formatting, like so: 

<img width="758" alt="Screenshot 2025-01-30 at 10 27 04 AM" src="https://github.com/user-attachments/assets/16e35876-c6f3-4f7d-b6e5-7ef37e11938f" />

> * Add a partition by attribute to specify the scope of sort node. :pr:`24095`. (`#24095 <https://github.com/prestodb/presto/pull/24095>`_)

Reviewer: ```suggestion
* Add a partition by attribute to specify the scope of sort node. `#24095 <https://github.com/prestodb/presto/pull/24095>`_
```

> * Remove ConnectorJoinNode from the SPI. JoinNode can now be used instead. :pr:`23976`. (`#23976 <https://github.com/prestodb/presto/pull/23976>`_)

Reviewer: ```suggestion
* Remove ConnectorJoinNode from the SPI. JoinNode can now be used instead. `#23976 <https://github.com/prestodb/presto/pull/23976>`_
```

> * Remove experimental getPreferredShuffleLayout methods from the connector metadata in favor of existing `getNewTableLayout`, `getInsertLayout` methods :pr:`12345`. (`#24106 <https://github.com/prestodb/presto/pull/24106>`_)

Reviewer: ```suggestion
* Remove experimental getPreferredShuffleLayout methods from the connector metadata in favor of existing `getNewTableLayout`, `getInsertLayout` methods `#24106 <https://github.com/prestodb/presto/pull/24106>`_
```

> * Modify the signature of ``PlanCheckerProviderFactory.create`` to pass in a map of configuration properties and replace ``SimplePlanFragmentSerde`` with a ``PlanCheckerProviderContext``. :pr:`23955`. (`#23955 <https://github.com/prestodb/presto/pull/23955>`_)

Reviewer: ```suggestion
* Modify the signature of ``PlanCheckerProviderFactory.create`` to pass in a map of configuration properties and replace ``SimplePlanFragmentSerde`` with a ``PlanCheckerProviderContext``. `#23955 <https://github.com/prestodb/presto/pull/23955>`_
```

> * Improve partitioned remote exchanges for wide data sets (more than 500 columns) to use row wise encoding. :pr:`23929 `. (`#23929 <https://github.com/prestodb/presto/pull/23929>`_)

Reviewer: ```suggestion
* Improve partitioned remote exchanges for wide data sets (more than 500 columns) to use row wise encoding. `#23929 <https://github.com/prestodb/presto/pull/23929>`_
```

> * Add native plan checker to the native sidecar plugin and native endpoint for Velox plan conversion. :pr:`23596`. (`#23596 <https://github.com/prestodb/presto/pull/23596>`_)

Reviewer: ```suggestion
* Add native plan checker to the native sidecar plugin and native endpoint for Velox plan conversion. `#23596 <https://github.com/prestodb/presto/pull/23596>`_
```

> * Add session properties ``native_spill_prefixsort_enabled``, ``native_prefixsort_normalized_key_max_bytes ``, and ``native_prefixsort_min_rows ``. :pr:`24043`. (`#24043 <https://github.com/prestodb/presto/pull/24043>`_)

Reviewer: ```suggestion
* Add session properties ``native_spill_prefixsort_enabled``, ``native_prefixsort_normalized_key_max_bytes ``, and ``native_prefixsort_min_rows ``. `#24043 <https://github.com/prestodb/presto/pull/24043>`_
```

**Details**
===========

General Changes
_______________
* Fix behavior of :func:`width_bucket(x, bins) -> bigint` which previously treated all ``null`` elements in bins as ``0``. The function throws an error if it finds a ``null`` or non-finite element in ``bins``. `#24103 <https://github.com/prestodb/presto/pull/24103>`_
* Fix bug in ``strrpos`` function for multibyte characters. `#24226 <https://github.com/prestodb/presto/pull/24226>`_
* Improve Request Headers in the Authentication Filter Class. `#23380 <https://github.com/prestodb/presto/pull/23380>`_
* Improve coordinator task management performance. `#24369 <https://github.com/prestodb/presto/pull/24369>`_
* Improve efficiency of coordinator when running a large number of tasks. `#24288 <https://github.com/prestodb/presto/pull/24288>`_
* Improve scheduling for CTE materialization: Now, only the stages containing CTE table scans that reference CTE table write stages are blocked till the write is complete, instead of the entire query being blocked as was the case previously. This is controlled by the session property ``enhanced_cte_scheduling_enabled`` (on by default). `#24108 <https://github.com/prestodb/presto/pull/24108>`_

> * Improve HTTP responses from async page transport servlet, by sanitizing the URI before use. `#24004 <https://github.com/prestodb/presto/pull/24004>`_

Reviewer 4: I would try to think of something that is more understandable to the user.  No one will understand what an async page transport servlet is.

Reviewer: LGTM! (edited "Improve security of internal HTTP endpoints by sanitizing headers before they are used. ")

* Add :doc:`/sql/alter-table` SET PROPERTIES statement. `#21495 <https://github.com/prestodb/presto/pull/21495>`_
* Add :func:`google_polyline_decode` function to convert Google polyline to Presto ST_Geometry types. `#23999 <https://github.com/prestodb/presto/pull/23999>`_
* Add :func:`google_polyline_encode` function to convert Presto ST_Geometry to Google polyline types. `#23999 <https://github.com/prestodb/presto/pull/23999>`_

> * Add Delete, TableWriter, and TableFinish node to SPI. `#24088 <https://github.com/prestodb/presto/pull/24088>`_

Reviewer 4: Move to SPI section

> * Add SemiJoin, Join, TableWriter, Delete, and TableFinish node to connector optimizer. `#24154 <https://github.com/prestodb/presto/pull/24154>`_

Reviewer 4: Move to SPI

Reviewer: LGTM! (move)

> * Add ``ClientRequestFilter.java`` interface in Presto-spi. `#23380 <https://github.com/prestodb/presto/pull/23380>`_

Reviewer 4: ```suggestion
* Add ``ClientRequestFilter` . `#23380 <https://github.com/prestodb/presto/pull/23380>`_
```

Link to docs which were added for this interface.

Reviewer: That doc link should be 
```
:doc:`/develop/client-request-filter`
```

Reviewer 3: Fixed

* Add a configuration property ``plan-checker.config-dir`` to set the configuration directory for PlanCheckerProvider configurations. `#23955 <https://github.com/prestodb/presto/pull/23955>`_
* Add a session property ``include_values_node_in_connector_optimizer`` to enable connector optimizer optimize plans with values node.  `#24227 <https://github.com/prestodb/presto/pull/24227>`_

> * Add an optional input distribution constraint to DeleteNode. `#24104 <https://github.com/prestodb/presto/pull/24104>`_

Reviewer 4: I think we can remove this, it's not user facing AFAIA

Reviewer: LGTM! (deletion)

* Add ``native_enforce_join_build_input_partition`` session property to not enforce input partition for join build. `#24163 <https://github.com/prestodb/presto/pull/24163>`_
* Add catalog and schema level access checks in :doc:`/sql/use` statement. `#24182 <https://github.com/prestodb/presto/pull/24182>`_
* Add delete node in subfield pruning optimizer. `#24206 <https://github.com/prestodb/presto/pull/24206>`_
* Add single worker execution. To improve latency of tiny queries running on a large cluster, we introduce single worker execution mode: query will only use one node to execute and plan would be optimized accordingly. This feature can be turned on by the configuration property ``single-node-execution-enabled`` or the session property ``single_node_execution_enabled``. `#24172 <https://github.com/prestodb/presto/pull/24172>`_
* Add support for ORC metadata cache invalidation based on file modification time. `#24346 <https://github.com/prestodb/presto/pull/24346>`_
* Add support for ``ALTER VIEW RENAME TO`` operation, including the necessary infrastructure for connector implementations. `#23749 <https://github.com/prestodb/presto/pull/23749>`_

> * Add support for ``BigInt`` data type in the SQL Client on Presto UI on supported browsers. See `compatibility <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#browser_compatibility>`_ for the supported browsers. `#24336 <https://github.com/prestodb/presto/pull/24336>`_

Reviewer 4: I would move this to a new section for the UI.

Reviewer: LGTM! (move)

* Add support for the histogram statistic type. `#22365 <https://github.com/prestodb/presto/pull/22365>`_

> * Add support for time type partitioning in the ORC file format for Iceberg. `#24091 <https://github.com/prestodb/presto/pull/24091>`_

Reviewer 4: Iceberg section

Reviewer: LGTM! (move)

* Upgrade Jackson dependencies to 2.15.4 in response to `PRISMA-2023-0067 <https://www.ibm.com/support/pages/security-bulletin-vulnerability-jackson-core-might-affect-ibm-business-automation-workflow-prisma-2023-0067>`_. `#23753 <https://github.com/prestodb/presto/pull/23753>`_
* Upgrade snakeyaml to 2.0 in response to `CVE-2022-1471 <https://nvd.nist.gov/vuln/detail/CVE-2022-1471>`_ and `CVE-2022-25857 <https://nvd.nist.gov/vuln/detail/cve-2022-25857>`_ and `CVE-2017-18640 <https://nvd.nist.gov/vuln/detail/cve-2017-18640>`_ and `CVE-2022-38752 <https://nvd.nist.gov/vuln/detail/CVE-2022-38752>`_ and `CVE-2022-38751 <https://nvd.nist.gov/vuln/detail/CVE-2022-38751>`_ and `CVE-2022-38750 <https://nvd.nist.gov/vuln/detail/CVE-2022-38750>`_ and `CVE-2022-38749 <https://nvd.nist.gov/vuln/detail/CVE-2022-38749>`_ and `CVE-2022-41854 <https://nvd.nist.gov/vuln/detail/CVE-2022-41854>`_. `#24099 <https://github.com/prestodb/presto/pull/24099>`_
* Upgrade nanoid to 3.3.8 in response to `CVE-2024-55565 <https://www.cve.org/CVERecord?id=CVE-2024-55565>`_. `#24273 <https://github.com/prestodb/presto/pull/24273>`_

BigQuery Connector Changes
__________________________

> * Fix for BigQuery ``SELECT``. `#23957 <https://github.com/prestodb/presto/pull/23957>`_

Reviewer 4: This needs something a little more descriptive

Reviewer: LGTM! (rephrase)

Cassandra Connector Changes
___________________________
* Improve cryptographic protocol in response to `Weak SSL/TLS protocols should not be used <https://sonarqube.ow2.org/coding_rules?open=java%3AS4423&rule_key=java%3AS4423>`_. `#24436 <https://github.com/prestodb/presto/pull/24436>`_

Clickhouse Connector Changes
____________________________
* Add ``DateTime64`` type support. `#24344 <https://github.com/prestodb/presto/pull/24344>`_

Delta Connector Changes
_______________________
* Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. `#23401 <https://github.com/prestodb/presto/pull/23401>`_

Hive Connector Changes
______________________
* Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. `#23401 <https://github.com/prestodb/presto/pull/23401>`_

Hudi Connector Changes
______________________
* Add ``catalog.system.invalidate_metastore_cache`` procedure to invalidate all, or portions of, the metastore cache. `#23401 <https://github.com/prestodb/presto/pull/23401>`_

Iceberg Connector Changes
_________________________

> * Improve redundant getTable calls in Iceberg Connector. `#24376 <https://github.com/prestodb/presto/pull/24376>`_

Reviewer 4: What about `Improve performance of scan planning in Iceberg connector`?

Reviewer: LGTM! (rephrase)

* Improve scheduling for CTE materialization: Now, only the stages containing CTE table scans that reference CTE table write stages are blocked till the write is complete, instead of the entire query being blocked as was the case previously. This is controlled by the session property ``enhanced_cte_scheduling_enabled`` (on by default). `#24108 <https://github.com/prestodb/presto/pull/24108>`_
* Improve security of internal HTTP endpoints by sanitizing headers before they are used. `#24004 <https://github.com/prestodb/presto/pull/24004>`_
* Add :doc:`/sql/alter-table` SET PROPERTIES statement. `#21495 <https://github.com/prestodb/presto/pull/21495>`_
* Add :func:`google_polyline_decode` function to convert Google polyline to Presto ST_Geometry types. `#23999 <https://github.com/prestodb/presto/pull/23999>`_
* Add :func:`google_polyline_encode` function to convert Presto ST_Geometry to Google polyline types. `#23999 <https://github.com/prestodb/presto/pull/23999>`_
* Add Delete, TableWriter, and TableFinish node to SPI. `#24088 <https://github.com/prestodb/presto/pull/24088>`_

> * Add ``ClientRequestFilter.java`` interface in Presto-spi: :doc:`/develop/client-request-filter. `#23380 <https://github.com/prestodb/presto/pull/23380>`_

Reviewer: ```suggestion
* Add ``ClientRequestFilter.java`` interface in Presto-spi: :doc:`/develop/client-request-filter`. `#23380 <https://github.com/prestodb/presto/pull/23380>`_
```

The `/develop/client-request-filter` path needs. a closing `. Local doc build currently looks like this: 
<img width="775" alt="Screenshot 2025-02-10 at 10 16 48 AM" src="https://github.com/user-attachments/assets/c811c1ac-0bf6-4dfb-89c6-a7d01ead249e" />

Reviewer 3: Thanks steve! Fixes
